log.py

```diff
@@ -43,18 +44,20 @@ def format(self, record):
             data['traceback'] = self.formatException(record.exc_info)
         return json.dumps(data)
 
-# TODO PYTHON 3  In Python 3 we need to go the other way -- all logging
-# output must be Unicode.
-class UTF8Formatter(logging.Formatter):
-    """Encode all Unicode output to UTF-8 to prevent encoding errors."""
+
+class StringFormatter(logging.Formatter):
+    """Encode all output as a string.
+    
+    In Python 2, this means a UTF-8 bytestring. In Python 3, it means a
+    Unicode string.
+    """
     def format(self, record):
         try:
-            data = super(UTF8Formatter, self).format(record)
+            data = super(StringFormatter, self).format(record)
         except Exception, e:
-            data = super(UTF8Formatter, self).format(record)
-        if isinstance(data, unicode):
-            data = data.encode("utf8")
-        return data
+            data = super(StringFormatter, self).format(record)
```

Comment:
This isn't related to your change, but I'm not sure what this try-except could be for. It's catching an exception and then doing the exact same thing.

Reply:
That's true. There's also a place in JSONFormatter where we catch an exception and immediately re-raise it.